docs/src/man/joins.md

```diff
@@ -34,41 +53,171 @@ There are seven kinds of joins supported by the DataFrames package:
 
 You can control the kind of join that `join` performs using the `kind` keyword argument:
```

Comment:
Since the output is now longer, it would be useful to list the supported values before illustrating them. Maybe link to a description of the joins terminology (e.g. on Wikipedia) if that's not already the case.